lambdas/functions/gdr_s3_data_sharing/gdr_s3_data_sharing.py

```diff
@@ -70,15 +70,13 @@ def handler(event, context):
     batch_job_list = []
     curr_datetimestamp = util.get_datetimestamp()
     for source_s3_key in source_s3_key_list:
-        filename = s3.get_s3_filename_from_s3_key(source_s3_key)
-
         # Construct destination S3 key
         destination_s3_key_prefix = (
             f"{destination_s3_key_prefix}/" if destination_s3_key_prefix else ""
         )
-        destination_timestamp_prefix = f"{curr_datetimestamp}/"  # Adding trailing / for
+        destination_timestamp_prefix = f"{curr_datetimestamp}/"  # Adding trailing /
```

Comment:
Why do we need the destination timestamp?
To have a dedicated prefix for each sharing request?

Reply:
Yes, I am thinking so it won't overlap/overwrite with possible existing shared data. I am assuming they could process/write files back to same directory, and separating it would avoid any mix up.

Reply:
Yup, I understand.
For a simple data copy out strategy that works fine. 
I does make a sync of the data tricky though. Even if you knew the timestamp, the function would generate its own and copy to a new location, right?

Say we share 1000 files and one is not transferred. It would be impossible to "add" that file to the first batch as the timestamp would be different, right?

There are a few options we could consider:
- check that the prefix does not exist and fail/warn if it does and it's supposed to be a new sharing process
- don't care about overwriting as it's in the responsibility of the receiver (as long as we make sure not to overwrite anything within our sharing session)
- put the timestamp as an input to the method, so it can be defined (if required, e.g. for syncs/reruns)
- don't use a timestamp at all and assume/require that the destination bucket has the same layout as our store bucket (if the receiver messes with that, he'll do so on his own risk)

Reply:
Yup, updated the code to include a manually generated timestamp prefix as one of the input. I suppose this could enable to patch data to the same sharing request.

Also update the code to check destination bucket keys, and filter out which key has exist in the destination bucket.